### PR TITLE
🌱 Drop support for INIT env variables in clusterctl upgrade test

### LINKED ISSUE #8535: Get rid of the clusterctl upgrade test fields exposed as environment variables
### What steps did you take and what happened?

Currently, clusterctl upgrade tests exposes multiple fields from `ClusterctlUpgradeSpecInput` as environment variable. Those env vars are only used by other infra providers and they can always read an env var in their `_test.go` file if they want to do that.

### What did you expect to happen?

Not to expose the fields as env vars. Please see https://github.com/kubernetes-sigs/cluster-api/pull/8518#discussion_r1166789179 for more info

### Cluster API version

main and supported release branches

### Kubernetes version

_No response_

### Anything else you would like to add?

_No response_

### Label(s) to be applied

/kind bug
One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels.

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8535

### CODE REVIEW

File: test/e2e/clusterctl_upgrade.go [hunk 1]
```diff
@@ -210,34 +202,22 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
...
-		if input.E2EConfig.HasVariable(initWithProvidersContract) {
-			initContract = input.E2EConfig.GetVariable(initWithProvidersContract)
-		}
 		if input.InitWithProvidersContract != "" {
 			initContract = input.InitWithProvidersContract
 		}
 
 		initKubernetesVersion = input.InitWithKubernetesVersion
-		if initKubernetesVersion == "" {
-			Expect(input.E2EConfig.Variables).To(HaveKey(initWithKubernetesVersion), "Invalid argument. %s variable must be defined when calling %s spec", initWithKubernetesVersion, specName)
```
Comment (chrischdi) [inline test/e2e/clusterctl_upgrade.go:237]: Should we do a `Expect(initKubernetesVersion).ToNot(BeEmpty())` here now?

File: test/e2e/clusterctl_upgrade.go [hunk 2]
```diff
@@ -210,34 +202,22 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		Expect(input.E2EConfig).ToNot(BeNil(), "Invalid argument. input.E2EConfig can't be nil when calling %s spec", specName)
 		Expect(input.ClusterctlConfigPath).To(BeAnExistingFile(), "Invalid argument. input.ClusterctlConfigPath must be an existing file when calling %s spec", specName)
 		Expect(input.BootstrapClusterProxy).ToNot(BeNil(), "Invalid argument. input.BootstrapClusterProxy can't be nil when calling %s spec", specName)
+		Expect(input.InitWithBinary).ToNot(BeEmpty(), "Invalid argument. input.InitWithBinary can't be empty when calling %s spec", specName)
+		Expect(input.InitWithKubernetesVersion).ToNot(BeEmpty(), "Invalid argument. input.InitWithKubernetesVersion can't be empty when calling %s spec", specName)
 
 		clusterctlBinaryURLTemplate := input.InitWithBinary
-		if clusterctlBinaryURLTemplate == "" {
-			Expect(input.E2EConfig.ResolveReleases(ctx)).To(Succeed(), "Failed to resolve release markers in e2e test config file")
```
Comment (chrischdi) [inline test/e2e/clusterctl_upgrade.go:216]: Same here?

**Review by fabriziopandini — COMMENTED** [review]
Nice!
/lgtm
/approve

**Pull request merged**